Dockerfile.buildkit

```diff
@@ -0,0 +1,59 @@
+# syntax = docker/dockerfile-upstream:1.0.1-experimental
```

Comment:
Could we use a template or something to avoid duplication?

It would mean the Dockerfile is generated dynamically which is nasty but duplication is asking for issues.

Reply:
As discussed offline, let's leave this for when we migrate to Jenkins. We can then require BuildKit and remove the other Dockerfile.